src/node_crypto.cc

```diff
@@ -741,6 +743,64 @@ void SecureContext::AddCRL(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void SecureContext::UseExtraCaCerts(const char* fname) {
+  extra_root_certs_file = fname;
+}
+
+static void emitWarning(Environment* env, const char* fmt, ...) {
+  char warning[1024];
+  va_list ap;
+
+  va_start(ap, fmt);
+  vsnprintf(warning, sizeof(warning), fmt, ap);
+  va_end(ap);
+
+#if 0
```

Comment:
@bnoordhuis what is "internal JS code"? In this case I am calling https://github.com/nodejs/node/blob/6845d6e3189cb6f7d2c847ab34a4f5b2eaf7a77f/lib/internal/process/warning.js#L28-L49, which is internal. It does nextTick a `process.emit()`, which can call user-code. Is that the difference here? That user code (`process.on('warning', user....`) was triggered from within `createServer()`, so we need to use `MakeCallback()` to keep the accounting correct for domains, etc.?

Otherwise, this doesn't seem so different from url.js, which calls into c++ to do some work, and synchronously calls back with `->Call()` to internal js code. Or node_http_parser, which uses `Call()` to call `pushValueToArray()`, a js function from `lib/internal/bootstrap_node.js`

I (and @mhdawson , perhaps others), would like to understand the basic principle here, so I correctly distinguish between the two in the future.

Also, with `MakeCallback()`:
```
require("tls").createServer()
  // -> MakeCallback
  //   -> process.emitWarning()
  //   ->  env->isolate()->RunMicrotasks()
  //   -> env->tick_callback_function()->Call()
  // return from createServer()
```

How would I demonstrate that `createServer()`, when it uses `MakeCallback()`, is causing micro tasks and ticks to get called? Or do they not? I'm trying to see what the side-effects are, but couldn't prove there were any with https://gist.github.com/sam-github/927a451526de55847d53694b53b01a20 (I was looking to cause output between the `BEGIN` and `END`, which should not happen).


I addressed your other comments, thanks, and I will fix the handle leak, looking into that now.